### PR TITLE
Solve problems

### PR DESCRIPTION
**Pull request by wolfv**
This PR parses solver problems into a vector of enums. We need to post process that list of problems into a graph, then compress it and print it to get similar error messages as with `libmamba` :)

### CODE REVIEW

File: crates/rattler_solve/src/libsolv/wrapper/pool.rs [hunk 1]
```diff
@@ -196,6 +196,21 @@ impl Pool {
             (id != 0).then_some(StringId(id))
         }
     }
+
+    /// Returns a list of solvables that match the provided id (id can be created from matchspec)
+    pub fn select_by_dep_id(&self, id: Id) -> Vec<SolvableId> {
```
Comment (baszalmstra) [inline crates/rattler_solve/src/libsolv/wrapper/pool.rs:201]: Is this a MatchSpecId? Maybe we can use that type here?

File: crates/rattler_solve/src/libsolv/wrapper/solver.rs [hunk 2]
```diff
@@ -87,6 +150,10 @@ impl Solver<'_> {
             // Safe because we know the `transaction` ptr is valid
             Ok(unsafe { Transaction::new(self, transaction) })
         } else {
+            let problems = self.all_solver_problems();
+            println!("problems: {:#?}", problems);
```
Comment (baszalmstra) [inline crates/rattler_solve/src/libsolv/wrapper/solver.rs:154]: Left over `println`

File: crates/rattler_solve/Cargo.toml [hunk 3]
```diff
@@ -13,6 +13,8 @@ tracing = "0.1.29"
 serde = { version = "1.0.130", features = ["derive"] }
 url = "2.2.2"
 hex = "0.4.3"
+graphlib = "0.6.3"
+petgraph = "0.6.3"
```
Comment (baszalmstra) [inline crates/rattler_solve/Cargo.toml:17]: These don't appear to be used yet?

Comment (wolfv) [reply crates/rattler_solve/Cargo.toml]: removed!

File: crates/rattler_solve/src/libsolv/wrapper/solver.rs (hunk 2 above)
Comment (wolfv) [reply crates/rattler_solve/src/libsolv/wrapper/solver.rs:155]: yeah I used it for testing. Shoudl we keep the PR WIP and leave it in, or try to merge it already?

File: crates/rattler_solve/src/libsolv/wrapper/pool.rs (hunk 1 above)
Comment (wolfv) [reply crates/rattler_solve/src/libsolv/wrapper/pool.rs:201]: Nope, unfortunately it's something else (a `relid` or `depid` ... but maybe we should make a new type for it on the Rust side).

Comment (baszalmstra) [reply crates/rattler_solve/src/libsolv/wrapper/pool.rs:201]: Yeah lets :)

File: crates/rattler_solve/src/libsolv/wrapper/solver.rs (hunk 2 above)
Comment (baszalmstra) [reply crates/rattler_solve/src/libsolv/wrapper/solver.rs:155]: Wip is fine 👍

Comment (wolfv) [reply crates/rattler_solve/src/libsolv/wrapper/solver.rs]: I removed it for now

File: crates/rattler_solve/src/libsolv/wrapper/pool.rs (hunk 1 above)
Comment (wolfv) [reply crates/rattler_solve/src/libsolv/wrapper/pool.rs]: I removed this function for the time being and would rather implement it similar to how we do String Id and friends

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**